InnerEye/ML/run_ml.py

```diff
@@ -669,3 +676,29 @@ def run_model_test(data_split: ModelExecutionMode) -> Optional[InferenceMetrics]
                         train_metrics=train_metrics, run_context=run_context)  # type: ignore
 
         return test_metrics, val_metrics, train_metrics
+
+    def download_weights(self) -> Path:
+        target_folder = self.project_root / fixed_paths.MODEL_WEIGHTS_DIR_NAME
+        target_folder.mkdir(exist_ok=True)
+
+        url = self.model_config.weights_url
+
+        # assign the same filename as in the download url if possible, so that we can check for duplicates
+        # If that fails, map to a random uuid
+        file_name = os.path.basename(requests.utils.urlparse(url).path) or str(uuid.uuid4().hex)
```

Comment:
does that work fine with SAS tokens in the URL?

Reply:
Yes, it does. urllib parses the URL into the netloc, path, query etc, so a SAS token in the URL won't be a problem